src/lxml/html/html5parser.py

```diff
@@ -25,12 +24,41 @@
 except ImportError:
     from urllib.parse import urlparse
 
+def _dodgeUseChardet(method_name):
+    # html5lib does not accept useChardet as an argument, if it
+    # detected the html argument would produce unicode objects.
+    # However, there is no reasonable way to predict if html5lib will
+    # detect the argument to be unicode (all of that code is private),
+    # so we'll have to settle for a retry.
+
+    # this function creates a wrapper around the specified object, which
+    # retries when html5lib complains about the useChardet argument
+
+    def inner(self, *args, **kwargs):
+        callee = getattr(super(type(self), self), method_name)
```

Comment:
This will fail if the class gets subtyped. I think a mixin class would be better here.